### PR TITLE
DeveloperGuide.md: Add UI features

### LINKED ISSUE #77: Developer Guide for UI features
Add the `UI` features to the Developer Guide Documentation

### PR DESCRIPTION
**Pull request by isaacsaw25**
Add UI description and class diagram to Developer Guide
close #77

### CODE REVIEW

File: docs/UML_Diagrams/Untitled Diagram.drawio
Comment (ThisisXXZ) [inline docs/UML_Diagrams/Untitled Diagram.drawio:1]: You may change the name of this file or simply remove it?

Comment (isaacsaw25) [reply docs/UML_Diagrams/Untitled Diagram.drawio:1]: I think it was Paul's file and I'm not sure if he still needs it, so I just moved it to the diagrams folder

File: docs/UML_Diagrams/UIClass.drawio.svg
Comment (ThisisXXZ) [inline docs/UML_Diagrams/UIClass.drawio.svg:1]: Nice diagram! A minor flaw (correct me if I'm wrong): I believe that the arrow on the dashed line (from ``UI``, point to ``Student``) should be open (unfilled), different from the arrow on a solid line.

Comment (isaacsaw25) [reply docs/UML_Diagrams/UIClass.drawio.svg:1]: Good point! I've changed the arrowhead to an open one for less ambiguity.

**Comment by isaacsaw25** [comment]
@ThisisXXZ Thanks for the prompt review 🫡

**Review by ThisisXXZ — APPROVED** [review]
Ready to merge!

**Pull request merged**